lab_1_classify_by_unigrams/main.py

```diff
@@ -1,107 +1,61 @@
+        return None
+    if len(predicted) != len(actual):
+        return None
+    p = len(predicted)
+    difference_square = []
+    for i in range(p):
+        difference_square.append((actual[i] - predicted[i]) ** 2)
+    mse = sum(difference_square) / p
+    return mse
 
 
 def compare_profiles(
         unknown_profile: dict[str, str | dict[str, float]],
-        profile_to_compare: dict[str, str | dict[str, float]]
-) -> float | None:
-    """
-    Compares profiles and calculates the distance using symbols
-    :param unknown_profile: a dictionary of an unknown profile
-    :param profile_to_compare: a dictionary of a profile to compare the unknown profile to
-    :return: the distance between the profiles
-    """
-
-
-def detect_language(
-        unknown_profile: dict[str, str | dict[str, float]],
-        profile_1: dict[str, str | dict[str, float]],
-        profile_2: dict[str, str | dict[str, float]],
-) -> str | None:
-    """
-    Detects the language of an unknown profile
-    :param unknown_profile: a dictionary of a profile to determine the language of
-    :param profile_1: a dictionary of a known profile
-    :param profile_2: a dictionary of a known profile
-    :return: a language
-    """
-
-
-def load_profile(path_to_file: str) -> dict | None:
-    """
-    Loads a language profile
-    :param path_to_file: a path to the language profile
-    :return: a dictionary with at least two keys – name, freq
-    """
-
-
-def preprocess_profile(profile: dict) -> dict[str, str | dict] | None:
-    """
-    Preprocesses profile for a loaded language
-    :param profile: a loaded profile
-    :return: a dict with a lower-cased loaded profile
-    with relative frequencies without unnecessary ngrams
-    """
-
-
-def collect_profiles(paths_to_profiles: list) -> list[dict[str, str | dict[str, float]]] | None:
-    """
-    Collects profiles for a given path
-    :paths_to_profiles: a list of strings to the profiles
-    :return: a list of loaded profiles
-    """
-
-
-def detect_language_advanced(unknown_profile: dict[str, str | dict[str, float]],
-                             known_profiles: list) -> list | None:
-    """
-    Detects the language of an unknown profile
-    :param unknown_profile: a dictionary of a profile to determine the language of
-    :param known_profiles: a list of known profiles
-    :return: a sorted list of tuples containing a language and a distance
-    """
+        profile_to_compare: dict[str, str | dict[str, float]]) -> float | None:
+    if not isinstance(unknown_profile, dict) or not isinstance(profile_to_compare, dict):
+        return None
+    if not ('name' and 'freq') in unknown_profile or not ('name' and 'freq') in profile_to_compare:
+        return None
+    unknown_freq = list(unknown_profile['freq'].values())
+    to_compare_freq = list(profile_to_compare['freq'].values())
+    mse_to_compare = calculate_mse(unknown_freq, to_compare_freq)
```

Comment:
What if languages contain different numbers of tokens? How do you ensure that the lists passed to MSE have information about the same token at each position?

Reply:
I think I improved it a little